api/app/utils/task_metrics.py

```
@@ -0,0 +1,410 @@
+
+        except Exception as e:
+            # Calculate duration even on failure
+            duration = time.time() - start_time
+            FAQ_EXTRACTION_DURATION.observe(duration)
+
+            # Update failure metrics
+            FAQ_EXTRACTION_RUNS.labels(status="failure").inc()
+            FAQ_EXTRACTION_LAST_RUN_STATUS.set(0)
+
+            # Re-raise the exception
+            raise e
```

Ruff (0.14.4)

Use `raise` without specifying exception name

Remove exception name

(TRY201)